File: 01 - 시간 복잡도, 배열, 연결리스트/kare9981/[BOJ] 1475.py
# tc = input()

# ans = [0] * 10
# for i in range(len(tc)):
#     num = int(tc[i])
#     if num == 6 or num == 9:
#         if ans[6] <= ans[9]:
#             ans[6] += 1
#         else:
#             ans[9] += 1
#     else:
#         ans[num] += 1
        
# print(max(ans))

n = input()
arr = [0] * 10

for i in range(len(n)):
    arr[int((n[i]))] += 1
arr[6] = (arr[6] + arr[9] + 1) // 2
arr[9] = arr[6]

print(max(arr))

File: 01 - 시간 복잡도, 배열, 연결리스트/oijuice/[BOJ] 1475.cpp
#include <iostream>
#include <algorithm>
#include <string>
#include <vector>
using namespace std;

int main(void){
  string input;
  cin >> input;

  vector<int> num(10);
  for(int i=0;i<input.size();i++) num[input[i]-'0']++;

  int max_val=0;
  for(int i=0;i<10;i++){
    if(i!=9&&i!=6) max_val=max(max_val, num[i]);

  cout << max(max_val,(num[6]+num[9]+1)/2);
}